normalizer.go

```diff
@@ -0,0 +1,179 @@
+package sqllexer
+
+import (
+	"regexp"
+	"strings"
+)
+
+type SQLNormalizerConfig struct {
+	DBMS string `json:"dbms"`
+
+	// CollectTables specifies whether the normalizer should also extract the table names that a query addresses
+	CollectTables bool `json:"collect_tables"`
+
+	// CollectCommands specifies whether the normalizer should extract and return commands as SQL metadata
+	CollectCommands bool `json:"collect_commands"`
+
+	// CollectComments specifies whether the normalizer should extract and return comments as SQL metadata
+	CollectComments bool `json:"collect_comments"`
+
+	// KeepSQLAlias reports whether SQL aliases ("AS") should be truncated.
+	KeepSQLAlias bool `json:"keep_sql_alias"`
+}
+
+type NormalizedInfo struct {
```

Comment:
I wonder if we can find a better name for this. Since that's info extracted from the statement rather than tied to normalization per say, would `StatementInfo` work for this?

Reply:
Thinking to name it `StatementMetadata`

Reply:
I love it. Perfect.